components/match2/commons/starcraft_starcraft2/match_group_input_starcraft.lua

```diff
@@ -993,9 +991,9 @@ function StarcraftMatchGroupInput._processDefaultPlayerMapData(players, map, opp
 	)
 
 	for playerIndex = 1, #players do
-		local faction = string.lower(map['t' .. opponentIndex .. 'p' .. playerIndex .. 'race'] or '')
+		local faction = map['t' .. opponentIndex .. 'p' .. playerIndex .. 'race']
 		participants[opponentIndex .. '_' .. playerIndex] = {
-			faction = _FACTIONS[faction] or players[playerIndex].extradata.faction or 'u',
+			faction = Faction.read(faction or players[playerIndex].extradata.faction) or Faction.defaultFaction,
```

Comment:
The extradata field wasn't normalized in the previous version, but is here. Is it intended?

Reply:
```suggestion
			faction = Faction.read(faction) or players[playerIndex].extradata.faction or Faction.defaultFaction,
```
`players[playerIndex].extradata.faction` is already normalized so moving the `)` earlier is better

thx